### DISCUSSION TITLE
ProgressBar: Add Buffer mode (or Multiple Progress Bars)

### DISCUSSION THREAD

**Discussion by J43fura** [primefaces/community#1418] [PrimeVue]
### Requested Feature
A "buffer" mode to the ProgressBar component to display two layers of progress: the primary progress and a secondary buffer progress, enhancing feedback for operations such as loading, streaming, or preloading content.

### Description
The buffer mode is essential for scenarios requiring a visual representation of both current progress and the extent of additional processing or preloading, similar to how YouTube indicates video loading. 
This mode adds a secondary progress bar behind the primary one but different color/opacity to distinguish the buffered amount from the actual progress.
![image](https://github.com/primefaces/community/assets/73950268/5921ef86-a949-473c-9ffc-c1135d51177d)

### Proposed API Changes
- New Prop: `mode="buffer"`
- Additional Prop: `bufferValue`: Number representing the secondary progress.
- New ProgressBar Pass Through option: `buffer`

E.g.:
`<ProgressBar mode="buffer" :value="30" pt:value:class="bg-red-500" :bufferValue="60" pt:buffer:class="bg-red-200" />`
![image](https://github.com/primefaces/community/assets/73950268/65ab2213-b205-4e4a-a400-bead05b3de3f)

### Multiple Progress Bars
An even better feature would be "multiple progress bars" (1+) instead of just "buffer" (2) E.g. [Boostrap](https://getbootstrap.com/docs/4.0/components/progress/#multiple-bars)
![image](https://github.com/primefaces/community/assets/73950268/b804b65c-b413-4450-802f-097c4cc58013)

**Comment by tugcekucukoglu** [comment]
We need more demand for `buffer` mode. And also I want to suggest using [MeterGroup](https://primevue.org/metergroup/) for `Multiple Progress Bars`.

**Reply by jbland-riparian** [reply]
I need to have multiple bars like above as we are migrating from bootstrap-vue to PrimeVue. BUT I am using PrimeVue 2. Do you have a solution for multiple bars like above for PrimeVue 2
